tests/OnceTest.php

```diff
@@ -216,4 +216,12 @@ public function getNumber()
         $this->assertTrue(Cache::isEnabled());
         $this->assertEquals($testClass->getNumber(), $testClass->getNumber());
     }
+
+    /** @test */
+    public function it_will_not_throw_error_with_anonymous_functions()
+    {
+        $result = eval('return once( function () { return random_int(1, 1000); } ) ;');
+
+        $this->assertTrue(in_array($result, range(1, 1000)));
```

Comment:
Just to be sure, could you let the test generate two numbers and assert that they are equal?

Reply:
This Pull request only prevent the error from being thrown, and does not make once work with functions, thus the test will not be able to assure that the two numbers are equal.

Reply:
@freekmurze can you please let me know what do you think about it.